pennylane_lightning/lightning_tensor/_tensornet.py

```diff
@@ -0,0 +1,196 @@
+
+        self._device_name = device_name
+        self._tensornet = self._tensornet_dtype()(self._num_wires, self._max_bond_dim)
+
+    @property
+    def dtype(self):
+        """Returns the tensor network data type."""
+        return self._dtype
+
+    @property
+    def device_name(self):
+        """Returns the tensor network device name."""
+        return self._device_name
+
+    @property
+    def num_wires(self):
+        """Number of wires addressed on this device"""
+        return self._num_wires
+
+    @property
+    def tensornet(self):
+        """Returns a handle to the tensor network."""
+        return self._tensornet
+
+    def _tensornet_dtype(self):
+        """Binding to Lightning Managed tensor network C++ class.
+
+        Returns: the tensor network class
+        """
+        return TensorNetC128 if self.dtype == np.complex128 else TensorNetC64
+
+    def reset_state(self):
+        """Reset the device's initial quantum state"""
+        # init the quantum state to |00..0>
+        self._tensornet.reset()
+
+    def _apply_basis_state(self, state, wires):
+        """Initialize the quantum state in a specified computational basis state.
+
+        Args:
+            state (array[int]): computational basis state of shape ``(wires,)``
+                consisting of 0s and 1s.
+            wires (Wires): wires that the provided computational state should be
+                initialized on
+
+        Note: This function does not support broadcasted inputs yet.
+        """
+        # length of basis state parameter
+        n_basis_state = len(state)
+
+        if not set(state.tolist()).issubset({0, 1}):
+            raise ValueError("BasisState parameter must consist of 0 or 1 integers.")
+
+        if n_basis_state != len(wires):
+            raise ValueError("BasisState parameter and wires must be of equal length.")
+
+        self._tensornet.setBasisState(state)
+
+    def _apply_lightning(self, operations):
+        """Apply a list of operations to the quantum state.
+
+        Args:
+            operations (list[~pennylane.operation.Operation]): operations to apply
+
+        Returns:
+            None
+        """
+        tensornet = self._tensornet
+
+        # Skip over identity operations instead of performing
+        # matrix multiplication with it.
+        for operation in operations:
+            if isinstance(operation, qml.Identity):
+                continue
+            if isinstance(operation, Adjoint):
+                name = operation.base.name
+                invert_param = True
+            else:
+                name = operation.name
+                invert_param = False
+            method = getattr(tensornet, name, None)
+            wires = list(operation.wires)
+
+            if method is not None:  # apply specialized gate
+                param = operation.parameters
+                method(wires, invert_param, param)
```

Comment:
We can revisit it later in Q3 when adding `exact tn` support. I'm not worry about its impact on the performance too much for now since the main computation cost is SVD decompose for `mps`.